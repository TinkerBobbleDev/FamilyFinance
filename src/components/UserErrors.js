import React from "react";

const UserErrors = props => {
  return (
    <div>
      {props.errors.length > 0 &&
        props.errors.map(error => {
          return (
            <p style={{ color: "red" }}>
              {error.code} : {error.message}
            </p>
          );
        })}
    </div>
  );
};

export default UserErrors;
