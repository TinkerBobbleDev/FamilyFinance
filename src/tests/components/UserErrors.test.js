import React from "react";
// import ReactShallowRenderer from "react-test-renderer/shallow";
import UserErrors from "../../components/UserErrors";
import toJSON from "enzyme-to-json";
import { createShallow } from "@material-ui/core/test-utils";

describe("<UserErrors />", () => {
  let shallow;

  beforeAll(() => {
    shallow = createShallow();
  });

  it("should work", () => {
    const wrapper = shallow(
      <UserErrors errors={[{ code: "error", message: "generic error" }]} />
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
