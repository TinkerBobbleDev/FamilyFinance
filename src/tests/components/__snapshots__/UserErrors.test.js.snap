// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserErrors /> should work 1`] = `
<div
  style={
    Object {
      "position": "fixed",
      "top": "10%",
      "width": "100%",
    }
  }
>
  <p
    key="error"
    style={
      Object {
        "color": "red",
      }
    }
  >
    error
     : 
    generic error
  </p>
</div>
`;
